simple_netconf_client/gui/SimpleNetconfClient.py

```diff
@@ -421,6 +435,76 @@ def __init__(self):
         # Start mDNS-SD scanning in background.
         self.start_zeroconf_scanner()
 
+        # Setup syntax highlighter
+        self.lexer = XmlLexer()
+        self.load_syntax_style(self.cfg.get("syntax_style", "monokai"))
+        self.textbox.bind("<KeyRelease>", self.on_key_release)
+
+    def load_syntax_style(self, name):
+        self.syntax_tags = []
+        style = get_style_by_name(name)
+
+        for token, opts in style.list_styles():
+            kwargs = {}
+            if opts.get("color", None):
+                kwargs["foreground"] = "#" + opts["color"]
+            if opts.get("bgcolor", None):
+                kwargs["background"] = "#" + opts["bgcolor"]
+            if opts.get("underline", None):
+                kwargs["underline"] = opts["underline"]
+
+            self.textbox.tag_config(str(token), **kwargs)
+            self.syntax_tags.append(str(token))
+
+        kwargs = {}
+        if style.background_color:
+            kwargs["fg_color"] = style.background_color
+
+        text_color = self.textbox.tag_cget("Token", "foreground")
+        if text_color:
+            kwargs["text_color"] = text_color
+
+        self.textbox.configure(**kwargs)
+
+        select_color = style.highlight_color
+        if select_color:
+            self.textbox.tag_config("sel", background=select_color)
+
+        self.highlight_syntax()
+        self.cfg["syntax_style"] = name
+
+    def highlight_syntax(self):
+        start = "1.0"
+        data = self.textbox.get(start, END)
+        data_size = len(data)
+        if data_size > self.cfg["max_highlighting_size"]:
+            return
+
+        while data and data[0] == '\n':
+            start = self.textbox.index('%s+1c' % start)
+            data = data[1:]
+        self.textbox.mark_set('range_start', start)
+
+        for t in self.syntax_tags:
+            self.textbox.tag_remove(
+                t, start, "range_start +%ic" % len(data)
+            )
+
+        for token, content in lex(data, self.lexer):
+            self.textbox.mark_set("range_end", f"range_start + {len(content)}c")
+            for t in token.split():
+                self.textbox.tag_add(str(t), "range_start", "range_end")
+            self.textbox.mark_set("range_start", "range_end")
+
+    def on_key_release(self, _):
+        current_time = time.time()
+
+        if current_time - self.last_key_pressed < 1: 
+            return
+        
+        self.last_key_pressed = current_time
```

Comment:
What about using a debounce instead? Right now the last input might not get highlighted if it happened too quickly after the previous one. With a debounce it would only get slightly delayed.

Reply:
This is a valid concern, haven't really thought this completely through, I will look into this.

Reply:
@owallb I have implemented simple debouncer with 500ms debounce. Work smoothly for smaller files, for bigger files we shall consider implementation of partial syntax highlighting but that shall be a concern for a different PR. Please test and review changes. Thanks for Your contribution.

Reply:
Beautiful, nice work.